src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs

```diff
@@ -49,6 +58,65 @@ public AWSResourceQueryer(IAWSClientFactory awsClientFactory)
             _awsClientFactory = awsClientFactory;
         }
 
+        public async Task<StackResourceDetail> DescribeCloudFormationResource(string stackName, string logicalId)
+        {
+            var cfClient = _awsClientFactory.GetAWSClient<IAmazonCloudFormation>();
+
+            var resource = await cfClient.DescribeStackResourceAsync(new DescribeStackResourceRequest {
+                LogicalResourceId = logicalId,
+                StackName = stackName
+            });
+
+            return resource.StackResourceDetail;
+        }
+
+        public async Task<EnvironmentDescription> DescribeElasticBeanstalkEnvironment(string environmentId)
+        {
+            var beanstalkClient = _awsClientFactory.GetAWSClient<IAmazonElasticBeanstalk>();
+
+            var environment = await beanstalkClient.DescribeEnvironmentsAsync(new DescribeEnvironmentsRequest {
+                EnvironmentNames = new List<string> { environmentId }
+            });
+
+            if (!environment.Environments.Any())
+            {
+                throw new AWSResourceNotFoundException($"The elastic beanstalk environment '{environmentId}' does not exist.");
+            }
+
+            return environment.Environments.First();
+        }
+
+        public async Task<Amazon.ElasticLoadBalancingV2.Model.LoadBalancer> DescribeElasticLoadBalancer(string loadBalancerArn)
+        {
+            var elasticLoadBalancingClient = _awsClientFactory.GetAWSClient<IAmazonElasticLoadBalancingV2>();
+
+            var loadBalancers = await elasticLoadBalancingClient.DescribeLoadBalancersAsync(new DescribeLoadBalancersRequest
+            {
+                LoadBalancerArns = new List<string> { loadBalancerArn }
+            });
+
+            if (!loadBalancers.LoadBalancers.Any())
+            {
+                throw new AWSResourceNotFoundException($"The load balancer '{loadBalancerArn}' does not exist.");
+            }
+
+            return loadBalancers.LoadBalancers.First();
+        }
+
+        public async Task<S3Region> GetS3BucketLocation(string bucketName)
```

Comment:
I recommend returning a string instead of S3Region and also handling the conversation of us-east-1 and eu-west-1 to the proper aws region name. For USEast1 the S3Region value is set to "" and for EUWest1 the S3Region value is "EU". This is due to legacy reasons from the `GetBucketLocation` API. If you let the weird S3Region behavior escape this method it will probably cause bugs.

Reply:
Fixed